ChatClient.go

```diff
@@ -125,6 +150,8 @@ func (user *ChatClient) Part() {
 
 // Quit : Disconnects from the chat server
 func (user *ChatClient) Quit() {
+	// You never need to cast nil to a different pointer type before doing a comparison,
+	// you can just do: if user.channel != nil
```

Comment:
I'd have to go back and take a look, but I was casting here because the comparison wasn't working with `!= nil`. It's one of those things in the language that I didn't understand the cause of, but found a way around.

Reply:
Hmm okay, I can't find any good reason why it wouldn't work with just `nil` but this is probably an area where my knowledge is lacking. Out of interest, did you try comparing it to an empty channel `UserChannel{}`?